lib/action_view/component/base.rb

```diff
@@ -8,6 +8,7 @@ class Base < ActionView::Base
       include ActiveModel::Validations
       include ActiveSupport::Configurable
       include ActionView::Component::Previewable
+      include ActionView::Component::ContentAreas
```

Comment:
Does this need to be a module if it's only included on one class?  I think this can make it harder to read the code and understand what interactions are happening.

Reply:
I think is actually clearer to keep the pieces separate - as we do for Previewable. There is a slightly strange interaction where the Base class interrogates content_areas to decide what to do with the captured content. Maybe we can clean that up.

Reply:
I wonder if ContentAreas should be extracted to a separate class. It seems like a separate set of responsibilities.

Reply:
Not convinced that is practical given how closely intertwined in needs to be in terms of having access to the components attributes etc.